app/screens/reaction_list/index.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+
+import {getMissingProfilesByIds} from 'mattermost-redux/actions/users';
+import {getUserIdsFromReactions, makeGetReactionsForPost} from 'mattermost-redux/selectors/entities/posts';
+import {getCurrentUserId, makeGetProfilesByIdsAndUsernames} from 'mattermost-redux/selectors/entities/users';
+import {getTeammateNameDisplaySetting, getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import ReactionList from './reaction_list';
+
+function makeMapStateToProps() {
+    const getReactionsForPostSelector = makeGetReactionsForPost();
+    const getProfilesByIdsAndUsernames = makeGetProfilesByIdsAndUsernames();
+
+    return function mapStateToProps(state, ownProps) {
+        const allUserIds = getUserIdsFromReactions(state, ownProps.postId);
+
+        return {
+            allUserIds,
+            currentUserId: getCurrentUserId(state),
+            reactions: getReactionsForPostSelector(state, ownProps.postId),
+            teammateNameDisplay: getTeammateNameDisplaySetting(state),
+            theme: getTheme(state),
+            userProfiles: getProfilesByIdsAndUsernames(state, {allUserIds}) || [],
```

Comment:
Setting this to `[]` by default should be handled in the component if necessary, although I think that selector always returns a value, so this shouldn't be necessary.

Reply:
Right, I guess we need to revisit all our selectors to return the default values so that the consumer will not worry to handle that.

Reply:
That selector always returns an array, so setting a default still isn't necessary here